### PR TITLE
systemtap: 2016-09-16 -> 2017-02-04

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Old version does not work with kernel 4.9 which is now the default for nixos 17.03

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/profiling/systemtap/default.nix [hunk 1]
```diff
@@ -14,7 +14,12 @@ let
   stapBuild = stdenv.mkDerivation {
     name = "systemtap-${version}";
     src = fetchgit { inherit url rev sha256; };
-    buildInputs = [ elfutils pkgconfig gettext ];
+    buildInputs = [ elfutils pkgconfig gettext python2 pythonPackages.setuptools ];
+    postPatch = ''
+      substituteInPlace buildrun.cxx --replace \
+        'o << "EXTRA_CFLAGS += -I\"" << s.runtime_path << "\"" << endl;' \
+        'o << "EXTRA_CFLAGS += -I" << s.runtime_path  << endl;'
+    '';
```
Comment (bendlas) [inline pkgs/development/tools/profiling/systemtap/default.nix]: I was just about to submit a PR on this aswell, but your workaround here is more minimal (I fixed it by recreating a full kbuild dir).  Could you add a comment though? Something like `FIXME: Workaround for bug in kbuild, where quoted -I"/foo" flags would get mangled in out-of-tree kbuild dirs`

Comment (volth) [inline pkgs/development/tools/profiling/systemtap/default.nix]: OK, maybe this issue should be reported to upstream (it not yet).

**Comment by bendlas** [comment]
+1 on the PR

I'm in the process of composing a bug report for kbuild. I've got trouble with producing a test case, though. The most minimal thing, just a dummy kernel mod with `-I"/foo"` doesn't seem to produce the error. Any help welcome.

**Comment by bendlas** [comment]
I've just reported this to linux-kbuild@vger.kernel.org along with a script to reproduce the quoting issue. You should have received a copy, @volth

**Pull request merged**